src/compile/data/data.ts

```diff
@@ -316,5 +314,16 @@ export function assembleData(model: Model, data: VgData[]) {
     data.push.apply(data, tuDomain);
   }
 
+  // remove data sources that are not needed
```

Comment:
Is it possible to just not add them at all up front?  removing later is generally a bad pattern that we should avoid unless we really need to.   Plus, you are iterating over the entire `data` array that is an input parameter, which means you might remove data source that's not even generated by this `assembleData`.  This seems like a very risky pattern.  

I guess you're only adding this to remove raw?   I think even if raw doesn't have its own transform.  
There is a good reason that we might need it separately.  
For example, if source is aggregated by `O1` and `O2`.  To sort ordinal scale of `O1`, you still need the raw source even if you don't do any transform in raw. 

Even if I'm wrong about we shouldn't remove, it seems like if we move the posFilter up to before adding the raw one, we can check if data source already contains raw.